traffic_monitor/cache/astats.go

```diff
@@ -68,50 +69,60 @@ type Astats struct {
 	System AstatsSystem           `json:"system"`
 }
 
-func astatsParse(cacheName string, rdr io.Reader) (Statistics, map[string]interface{}, error) {
+func astatsParse(cacheName string, rdr io.Reader, pollCTX interface{}) (Statistics, map[string]interface{}, error) {
 	var stats Statistics
 	if rdr == nil {
 		log.Warnf("%s handle reader nil", cacheName)
 		return stats, nil, errors.New("handler got nil reader")
 	}
 
-	var astats Astats
-	json := jsoniter.ConfigFastest
-	if err := json.NewDecoder(rdr).Decode(&astats); err != nil {
-		return stats, nil, err
-	}
+	ctx := pollCTX.(*poller.HTTPPollCtx)
 
-	if err := stats.AddInterfaceFromRawLine(astats.System.ProcNetDev); err != nil {
-		return stats, nil, fmt.Errorf("Failed to parse interface line for cache '%s': %v", cacheName, err)
-	}
-	if inf, ok := stats.Interfaces[astats.System.InfName]; !ok {
-		return stats, nil, errors.New("/proc/net/dev line didn't match reported interface line")
-	} else {
-		inf.Speed = int64(astats.System.InfSpeed)
-		stats.Interfaces[astats.System.InfName] = inf
-	}
+	ctype := ctx.HTTPHeader.Get("Content-Type")
 
-	if load, err := LoadavgFromRawLine(astats.System.ProcLoadavg); err != nil {
-		return stats, nil, fmt.Errorf("Failed to parse loadavg line for cache '%s': %v", cacheName, err)
-	} else {
-		stats.Loadavg = load
-	}
+	if ctype == "text/json" || ctype == "text/javascript" || ctype == "" {
+		var astats Astats
+		json := jsoniter.ConfigFastest
+		if err := json.NewDecoder(rdr).Decode(&astats); err != nil {
+			return stats, nil, err
+		}
 
-	stats.NotAvailable = astats.System.NotAvailable
+		if err := stats.AddInterfaceFromRawLine(astats.System.ProcNetDev); err != nil {
+			return stats, nil, fmt.Errorf("Failed to parse interface line for cache '%s': %v", cacheName, err)
```

Comment:
This seems like as good of a time as ever to remove the capitalization from this error message.